src/Serilog.Sinks.RollingFile/Sinks/RollingFile/RollingFileSink.cs

```diff
@@ -64,16 +69,26 @@ public sealed class RollingFileSink : ILogEventSink, IDisposable
                               long? fileSizeLimitBytes,
                               int? retainedFileCountLimit,
                               Encoding encoding = null,
-                              bool buffered = false)
+                              bool buffered = false,
+                              TimeSpan? retainedFileAgeLimit = null)
```

Comment:
The age limit is currently added as an optional parameter at the end of the `RollingFileSink` constructor and the corresponding extension methods, to avoid breaking existing code (though it does break binary compatibility).

Maybe a more elegant alternative could be to add an overloaded constructor that takes in one or more `IRetentionPolicy` parameters instead of `retainedFileCountLimit` and `retainedFileAgeLimit`, and matching extension methods? I'm not sure if this would play nicely with the XML/JSON configuration wiring though. Would that require changes to the Serilog configuration infrastructure? Is it possible to inject that behavior somehow instead?